### PR TITLE
feat: Add SOFARPC Framework Plugin Support

### PR DESCRIPTION
**Pull request by oewang**
- Support for collecting client-to-server traces
- Supports collecting the following metrics for the method dimension: number of calls, number of failures, minimum, average, maximum execution time, etc.
- Support for SOFARPC framework greater than or equal to version 5.3.0

### CODE REVIEW

File: plugins/sofarpc/src/test/java/com/megaease/easeagent/plugin/sofarpc/interceptor/trace/future/SofaRpcResponseFutureTraceInterceptorTest.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+		Context context = EaseAgent.getContext();
+		sofaRpcConsumerTraceInterceptor.before(methodInfo, context);
+		sofaRpcConsumerTraceInterceptor.after(methodInfo, context);
+		methodInfo.setArgs(futureInvokeCallbackConstructTracingInterceptorArgs);
+		sofaRpcFutureInvokeCallbackConstructInterceptor.before(methodInfo, context);
+		Thread asyncThread = new Thread(new Runnable() {
+			@Override
+			public void run() {
+				methodInfo.setInvoker(futureInvokeCallbackConstructTracingInterceptorArgs[2]);
+				sofaRpcResponseFutureTraceInterceptor.after(methodInfo, context);
```
[the remark this answers is not in the document]
Comment (oewang) [reply plugins/sofarpc/src/test/java/com/megaease/easeagent/plugin/sofarpc/interceptor/trace/future/SofaRpcResponseFutureTraceInterceptorTest.java:128]: I have found the problem point and solved it.

**Review by landyking — APPROVED** [review]
LGTM

File: plugins/sofarpc/src/test/java/com/megaease/easeagent/plugin/sofarpc/interceptor/trace/callback/SofaRpcResponseCallbackTraceInterceptorTest.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+				Assert.assertNotNull(asyncContext);
+				RequestContext requestContext = asyncContext.get(SofaRpcCtxUtils.CLIENT_REQUEST_CONTEXT_KEY);
+				asyncContext.put(SofaRpcCtxUtils.CLIENT_REQUEST_CONTEXT_KEY, null);
+				requestContext.span().finish();
+				requestContext.scope().close();
+				try {
+					sofaRpcResponseCallbackTrace.onSofaException(sofaTimeOutException, sofaRequest.getMethod().getName(), sofaRequest);
+				} catch (Exception ignore) {
+					Assert.assertTrue(ignore instanceof NullPointerException);
+				}
```
Comment (observeralone) [inline plugins/sofarpc/src/test/java/com/megaease/easeagent/plugin/sofarpc/interceptor/trace/callback/SofaRpcResponseCallbackTraceInterceptorTest.java:181]:
```suggestion
				} catch (NullPointerException ignore) {
				    //Must be throw NullPointerException
				    return;
				}
				throw new RuntimeException("Must be throw NullPointerException");
```

File: plugins/sofarpc/src/test/java/com/megaease/easeagent/plugin/sofarpc/interceptor/trace/future/SofaRpcResponseFutureTraceInterceptorTest.java [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+				Assert.assertNotNull(asyncContext);
+				RequestContext requestContext = asyncContext.get(SofaRpcCtxUtils.CLIENT_REQUEST_CONTEXT_KEY);
+				requestContext.span().finish();
+				asyncContext.put(SofaRpcCtxUtils.CLIENT_REQUEST_CONTEXT_KEY, null);
+				methodInfo.setInvoker(futureInvokeCallbackConstructTracingInterceptorArgs[2]);
+				try {
+					sofaRpcResponseFutureTraceInterceptor.after(methodInfo, context);
+				} catch (Exception e) {
+					Assert.assertTrue(e instanceof NullPointerException);
+				}
```
Comment (observeralone) [inline plugins/sofarpc/src/test/java/com/megaease/easeagent/plugin/sofarpc/interceptor/trace/future/SofaRpcResponseFutureTraceInterceptorTest.java:178]:
```suggestion
				} catch (NullPointerException ignore) {
				    //Must be throw NullPointerException
				    return;
				}
				throw new RuntimeException("Must be throw NullPointerException");
```

**Review by observeralone — APPROVED** [review]

**Pull request merged**